### PR TITLE
Issue 39696: Search reference for file breaks when renaming container

### PR DESCRIPTION
**Pull request by Sigmonia**

### CODE REVIEW

File: core/src/org/labkey/core/webdav/DavController.java [hunk 1]
```diff
@@ -3895,7 +3900,7 @@ WebdavStatus doMethod() throws DavException, IOException
         }
     }
 
-    private void fireFileMovedEvent(WebdavResource dest, WebdavResource src)
+    private void fireFileMovedEvent(WebdavResource dest, WebdavResource src, Collection<? extends WebdavResource> movedChildren)
```
Comment (labkey-susanh) [inline core/src/org/labkey/core/webdav/DavController.java]:
```suggestion
    private void fireFileMovedEvent(WebdavResource dest, WebdavResource src, @Nullable Collection<? extends WebdavResource> movedChildren)
```

File: api/src/org/labkey/api/webdav/AbstractWebdavResource.java [hunk 2]
```diff
@@ -96,7 +96,7 @@ protected AbstractWebdavResource(Path folder, String name)
 
     protected AbstractWebdavResource(Path folder, String name, WebdavResolver resolver)
     {
-        super(folder, name, resolver);
+        this(folder.append(name), resolver);
```
Comment (labkey-susanh) [inline api/src/org/labkey/api/webdav/AbstractWebdavResource.java]: This is now bypassing the validateName check in AbstractResource.  Is there a reason we want to bypass that check?

File: api/src/org/labkey/api/webdav/AbstractWebdavResource.java [hunk 3]
```diff
@@ -96,7 +96,7 @@ protected AbstractWebdavResource(Path folder, String name)
 
     protected AbstractWebdavResource(Path folder, String name, WebdavResolver resolver)
     {
-        super(folder, name, resolver);
+        super(folder.append(name), resolver);
```
Comment (labkey-susanh) [inline api/src/org/labkey/api/webdav/AbstractWebdavResource.java]:
```suggestion
        super(folder, name, resolver);
```
Still not sure why this needs to change.  Am I looking at the wrong version of the file somehow?

**Review by labkey-susanh — CHANGES_REQUESTED** [review]

**Review by labkey-susanh — APPROVED** [review]

File: api/src/org/labkey/api/webdav/AbstractWebdavResource.java (hunk 3 above)
Comment (Sigmonia) [reply api/src/org/labkey/api/webdav/AbstractWebdavResource.java]: This was homeschooling fatigue. Intended to revert it.

**Review by labkey-susanh — APPROVED** [review]

**Pull request merged**